inference-engine/thirdparty/clDNN/src/program.cpp

```diff
@@ -422,10 +422,18 @@ void program::build_program(bool is_internal) {
     init_graph();
     { pre_optimize_graph(is_internal); }
     run_graph_compilation();
-    { post_optimize_graph(is_internal); }
-    prepare_memory_dependencies();
-    compile();
-    init_kernels();
+
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+#ifdef GPU_DEBUG_CONFIG
+    if (debug_config->dry_run.empty()) {
+#else
+    {
+#endif
+        { post_optimize_graph(is_internal); }
```

Comment:
If `DryRun` option is enabled, then we skip some graph optimizations in `post_optimize_graph` stage. Is such behavior expected?

Reply:
Oh, that was due to my ignorance :( Thanks for pointing out!